torchao/utils.py

```diff
@@ -419,12 +423,83 @@ def decorator(func):
             def wrapper(f, types, args, kwargs):
                 return func(f, types, args, kwargs)
 
-            cls._ATEN_OP_OR_TORCH_FN_TABLE[op] = wrapper
+            cls._ATEN_OP_OR_TORCH_FN_TABLE[cls][op] = wrapper
         return func
 
     return decorator
 
 
+def _implements_common_tensor_ops(cls):
```

Comment:
moved the op impls to a separate functions so we can choose to register these things only when "tensor_data_names" and "tensor_attribute_names" are defined